plugin-energy-estimation-tdp/src/transform.rs

```diff
@@ -0,0 +1,112 @@
+use core::f64;
+use std::{
+    clone, collections::HashMap, sync::{Arc, Mutex}, time::{SystemTime, UNIX_EPOCH},
+};
+
+use alumet::{
+    measurement::{AttributeValue, MeasurementBuffer, MeasurementPoint, WrappedMeasurementValue},
+    pipeline::{
+        elements::{error::TransformError, transform::TransformContext},
+        Transform,
+    },
+    resources::Resource,
+};
+
+use serde::de::value;
+
+use crate::Config;
+
+pub struct EnergyEstimationTdpTransform {
+    pub config: Config,
+    pub metrics: Arc<Mutex<super::Metrics>>,
+}
+
+impl EnergyEstimationTdpTransform {
+    /// Instantiates a new EnergyAttributionTransform with its private fields initialized.
+    pub fn new(config: Config, metrics: Arc<Mutex<super::Metrics>>) -> Self {
+        Self {
+            config,
+            metrics,
+        }
+    }
+
+}
+
+impl Transform for EnergyEstimationTdpTransform {
+    /// Applies the transform on the measurements.
+    fn apply(&mut self, measurements: &mut MeasurementBuffer, _ctx: &TransformContext) -> Result<(), TransformError> {
+        // Retrieve the pod_id and the rapl_id.
+        // Using a nested scope to reduce the lock time.
+        log::trace!("EZC: enter in apply transform function");
```

Comment:
I don't think that EZC is worldwide known

Reply:
yes of course !